src/logic/calculate.js

```diff
@@ -0,0 +1,21 @@
+import operate from './operate';
+
+const calculate = (btnName, dataObj) => {
+  let { total } = dataObj;
+  const { next, operation } = dataObj;
+  const ops = {
+    AC: 0,
+    '+/-': -1,
+    '%': 0.01,
+  };
+  if (btnName === '=' || btnName === '-' || btnName === 'X'
+    || btnName === '÷') {
+    total = operate(total, next, btnName);
+  }
+  if (btnName === 'AC' || btnName === '+/-' || btnName === '%') {
+    total = total * next * ops[btnName];
+  }
+  return { total, next, operation };
+};
```

Comment:
As this might be an approach that, in paper, its fine, we are deviating from the [Pathwright Requirements](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896094/) 
![image](https://user-images.githubusercontent.com/46115725/73175673-52845180-40d0-11ea-9ba1-c024a073f912.png)
- To expand on that requirement, it would be a good idea to approach this logical section as follows: We are going to mutate the data object properties, all of them, not just the total, think of it like this, since the display of the calculator and the input is going to be closely related to this, we have that btnName is going to be the users input right, for example: the user's input is the btnName 'AC' so our if statement should look something like this `if ( btnName === 'AC' ) { total: null, next: null, operation: null }`, so now we need to think how are we going to modify the data object properties taking into consideration that we are going to mutate the properties according to the users input, so we will have to consider, what will be mutates when a user starts inputting numbers, what will happen when a user inputs an operation, but folowing the logic stated above, take a user input(btnName) and modify the properties of data object accordingly